Welcome to TrioMix's documentation!
===================================
TrioMix is a bioinformatics tool to detect **intrafamilial contamination**, **chimerism**, **uniparental disomy** by investigating inheritance patterns of SNPs. TrioMix quantifies the deviation from Mendelian inheritance patterns by using maximum likelihood esstimation (MLE) inferred from the genotypes of parent-offspring trio. TrioMix can be used on both whole-genome sequencing (WGS) of trios or whole-exome sequencing (WES) of trios. 

TrioMix can be installed by cloning the `github <https://github.com/cjyoon/triomix>`_ or using the pre-built `docker <https://hub.docker.com/repository/docker/cjyoon/triomix>`_ image. 

Check out the :doc:`usage` section for further information, including how to :ref:`install` the project.

If you use Triomix in your analysis, please cite our work below.

**"Estimation of intrafamilial DNA contamination in family trio genome sequencing using Mendelian inconsistencies. Yoon et al., Genome Research (2022)"** `[link to paper] <https://doi.org/10.1101/gr.276794.122>`_


.. .. note::

..    This project is under active development.


Contents
--------

.. toctree::

   install
   introduction
   usage
   example
   plots
   faq
