Welcome to TrioMix's documentation!
===================================
TrioMix is a bioinformatics tool to detect **intrafamilial contamination**, **uniparental disomy**, and **chimerism** by investigating inheritance patterns of SNPs. To estimate contamination it uses maximum likelihood esstimation (MLE) to accurately quantify contamination within family members of the parent-child trio. TrioMix can be used on both whole-genome sequencing (WGS) of trios or whole-exome sequencing (WES) of trios. 

TrioMix can be installed by cloning the `github <https://github.com/cjyoon/triomix>`_ or using the pre-built `docker <https://hub.docker.com/repository/docker/cjyoon/triomix>`_ image. 

Check out the :doc:`usage` section for further information, including how to :ref:`install` the project.

If you use Triomix in your analysis, please cite our work published in XXXX.

"Estimation of familial DNA contamination using Mendelian inconsistencies in next generation sequencing of trios. Yoon et al., XXXXX (2022)"


.. note::

   This project is under active development. Readthedocs pages are still underconstruction. 


Contents
--------

.. toctree::

   install
   introduction
   usage
   example
   plots
   upd
   faq
